htlcswitch/link.go

```diff
@@ -2144,6 +2157,27 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			"ChannelPoint(%v): received error from peer: %v",
 			l.channel.ChannelPoint(), msg.Error(),
 		)
+
+	case *lnwire.Shutdown:
+		// We've received a Shutdown message from the remote peer.
```

Comment:
Not clear why the link needs to know about this message now.

Would think instead the peer goroutine get this message, then signals to the link to let it know once the coast is clear. After the coast is clear, then the peer starts the coop close state machine and resumes from there.